classy_vision/optim/adam.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, Tuple
+
+import torch.optim
+from classy_vision.generic.util import is_pos_float
+from classy_vision.optim.param_scheduler import (
+    ClassyParamScheduler,
+    build_param_scheduler,
+)
+
+from . import ClassyOptimizer, register_optimizer
+
+
+@register_optimizer("adam")
+class Adam(ClassyOptimizer):
+    def __init__(
+        self,
+        lr_scheduler: ClassyParamScheduler,
+        betas: Tuple[float, float],
+        eps: float,
+        weight_decay: float,
+        amsgrad: bool,
```

Comment:
Can we add the defaults from PyTorch over here?

Reply:
Sure, but I also notice some inconsistency in SGD and RMSProp since some of their parameters are missing PyTorch Defaults or type annotations or both

Reply:
Yeah, that's a good point. We need to revisit our old code to make sure there aren't any discrepancies.